src/storage/mod.rs

```diff
@@ -148,6 +150,62 @@ impl<T: Component> Drop for MaskedStorage<T> {
     }
 }
 
+#[derive(Derivative)]
+#[derivative(Default(bound = ""))]
+/// Associates the component's storage with its metadata.
+pub struct WrappedStorage<T: Component> {
```

Comment:
I don't see why we need such a type.

Reply:
I think it is better for implementation, otherwise I'd have to repeat myself for every usage of `get`, `get_mut`, `insert`, `remove`, etc.